guides/keras_cv/classification_with_keras_cv.py

```diff
@@ -0,0 +1,688 @@
+`RandAugment` is actually a set of 10 different augmentations:
+`AutoContrast`, `Equalize`, `Solarize`, `RandomColorJitter`, `RandomContrast`,
+`RandomBrightness`, `ShearX`, `ShearY`, `TranslateX` and `TranslateY`.
+At inference time, `num_augmentations` augmenters are sampled for each image,
+and random magnitude factors are sampled for each.
+These augmentations are then applied sequentially.
+
+KerasCV makes tuning these parameters easy using the `augmentations_per_image`
+and `magnitude` parameters!
+Let's take it for a spin:
+"""
+rand_augment = keras_cv.layers.RandAugment(
+    augmentations_per_image=3,
+    magnitude=0.3,
+    value_range=(0, 255),
+)
+augmenters += [rand_augment]
+
+image_batch = rand_augment(image_batch)
+keras_cv.visualization.plot_image_gallery(
+    image_batch,
+    rows=3,
+    cols=3,
+    value_range=(0, 255),
+    show=True,
+)
+
+"""
+Looks great; but we're not done yet!
+What if an image is missing one critical feature of a class?  For example, what
+if a leaf is blocking the view of a Cat's ear, but our classifier learned to
+classify cats simply by observing their ears?
+
+One easy approach to tackling this is to use `RandomCutout`, which randomly
+strips out a sub-section of the image:
+"""
+random_cutout = keras_cv.layers.RandomCutout(width_factor=0.4, height_factor=0.4)
+keras_cv.visualization.plot_image_gallery(
+    random_cutout(image_batch),
+    rows=3,
+    cols=3,
+    value_range=(0, 255),
+    show=True,
+)
+
+"""
+While this tackles the problem reasonably well, it can cause the classifier to
+develop responses to borders between features and black pixel areas caused by
+the cutout.
+
+[`CutMix`](https://arxiv.org/abs/1905.04899) solves the same issue; but by using
+a more complex (and effective!) technique.
+Instead of replacing the cut-out areas with black pixels, `CutMix` replaces
+these regions with regions of other images sampled from within your training
+set!
+Following this replacement, the image's classification label is updated to be a
+blend of the original image's class label, as well as the image that was
+overlaid into the cutout section's label.
+
+What does this look like in practice?  Let's check it out:
+"""
+cut_mix = keras_cv.layers.CutMix()
+# CutMix needs to modify both images and labels
+inputs = {"images": image_batch, "labels": label_batch}
+
+keras_cv.visualization.plot_image_gallery(
+    cut_mix(inputs)["images"],
+    rows=3,
+    cols=3,
+    value_range=(0, 255),
+    show=True,
+)
+"""
+Let's hold off from adding it to our augmenter for a minute - more on that
+soon!
+
+Next, let's look into `MixUp()`.
+Unfortunately, while `MixUp()` has been empirically shown to *substantially*
+improve both the robustness and the generalization of the trained model,
+it is not well-understood why such improvement occurs.
+A little alchemy never hurt anyone!
+
+![](https://i.imgur.com/d4ZZYvW.png)
```

Comment:
Love the image, but we should remove it since it's distracting and we likely don't have copyright.

Reply:
hahaha I generated it myself!  it is probably too distracting though :(